handler/EventHandler_test.go

```diff
@@ -0,0 +1,199 @@
+package handler
+
+import (
+	"math/rand"
+	"net/http"
+	"runtime/debug"
+	"testing"
+	"time"
+
+	"github.com/ONSdigital/dp-import-reporter/config"
+	"github.com/coocood/freecache"
+	. "github.com/smartystreets/goconvey/convey"
+)
+
+//Correct
+var cfg = &config.Config{
+	NewInstanceTopic: "event-reporter",
+	Brokers:          []string{"localhost:9092"},
+	ImportAPIURL:     "http://localhost:21800",
+	ImportAuthToken:  "FD0108EA-825D-411C-9B1D-41EF7727F465",
+	BindAddress:      ":22200",
+}
+
+//wrong ImportAPI
+var cfg1 = &config.Config{
+	NewInstanceTopic: "event-reporter",
+	Brokers:          []string{"localhost:9092"},
+	ImportAPIURL:     "http://localho:21800",
+	ImportAuthToken:  "FD0108EA-825D-411C-9B1D-41EF7727F465",
+	BindAddress:      ":22200",
+	// ImportAPIURL: "http://localhost:21800",
+}
+
+//wrong auth token
+var cfg2 = &config.Config{
+	NewInstanceTopic: "event-reporter",
+	Brokers:          []string{"localhost:9092"},
+	ImportAPIURL:     "http://localhost:21800",
+	ImportAuthToken:  "D0108EA-825D-411C-9B12-41EF7727F465",
+	BindAddress:      ":22200",
+}
+
+//correct
+var e = EventReport{
+	InstanceID: "a4695fee-f0a2-49c4-b136-e3ca8dd40476",
+	EventType:  "error",
+	EventMsg:   "Broken on something.",
+}
+
+//wrong instance
+var e1 = EventReport{
+	InstanceID: "a4695fee-f0a2-49c4-b136-e3ca8dd40612",
+	EventType:  "error",
+	EventMsg:   "Broken on something.",
+}
+var e2 = EventReport{
+	InstanceID: "a4695fee-f0a2-49c4-b136-e3ca8dd40612",
+	EventType:  "error",
+	EventMsg:   "Broken on ",
+}
+var e3 = EventReport{
+	InstanceID: "a4695fee-f0a2-49c4-b136-e3ca8dd40476",
+	EventType:  "error",
+	EventMsg:   "Broken on something." + string(rand.Int()),
+}
+var httpClient = &http.Client{}
+
+func TestCheckInstance(t *testing.T) {
+
+	Convey("Internal method which checks if the instance exists and returns the status and events ", t, func() {
+
+		_, _, err := e.checkInstance(httpClient, cfg1)
+		Convey("URL should not parse", func() {
+			So(err, ShouldNotBeNil)
+		})
+
+		state, events, err := e.checkInstance(httpClient, cfg)
+		Convey("Complete run through with 200 status response", func() {
+			So(err, ShouldBeNil)
+			So(state, ShouldNotBeNil)
+			So(events, ShouldNotBeNil)
+		})
+
+		state1, events1, err := e1.checkInstance(httpClient, cfg1)
+		Convey("Complete run through with incorrect instanceID", func() {
+			So(err, ShouldNotBeNil)
+			So(state1, ShouldEqual, "")
+			So(events1, ShouldNotBeNil)
+		})
+	})
+
+}
+
+func TestPutJobStatus(t *testing.T) {
+
+	Convey("Internal method which changes the job status ", t, func() {
+		cfg3, _ := config.Get()
```

Comment:
rather than cfg1 cfg2 cfg3 consider giving more meaningful names that describe why they're different? cfgBadAuth cfgBadURL etc